test_conformance/basic/test_async_strided_copy.cpp

```diff
@@ -215,6 +215,8 @@ int test_strided_copy(cl_device_id deviceID, cl_context context, cl_command_queu
                 sprintf(values + strlen( values), "%2x ", outchar[j]);
             sprintf(values + strlen(values), "]");
             log_error("%s\n", values);
+            free(inBuffer);
```

Comment:
These two allocations could also leak from any early return of `test_error`. Might be better to replace with a vector instead so that there is no chance of leaking from an early return.

Reply:
Well, it seems most of the cases have the same problems.
If using `malloc`, might be better to use `goto` in `test_error` or replace all with `vector`.

Reply:
Yeah a lot of CTS code doesn't handle error cases well. What @Jeremy-Kemp is suggesting is better but I'd be fine with accepting this patch as is since it fixes the issue for the case where the test runs successfully.

Reply:
Agree we should use vectors but probably better to do it for all tests. I've created https://github.com/KhronosGroup/OpenCL-CTS/issues/1033.